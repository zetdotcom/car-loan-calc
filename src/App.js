import React, { useState } from 'react';
import FormAndSummaryContainer from 'containers/FormAndSummaryContainer';
import CarsListContainer from 'containers/CarsListContainer';
import Header from 'components/Header';
import { GlobalStyle } from './style';
import styled from 'styled-components';

const AppWrapper = styled.div`
	height: 100vh;
	overflow: auto;
`;

function App() {
	const [monthly, setMonthly] = useState('');

	return (
		<AppWrapper>
			<GlobalStyle />
			<Header />
			<FormAndSummaryContainer setMonthly={setMonthly} />
			<CarsListContainer monthly={monthly} />
		</AppWrapper>
	);
}

export default App;
