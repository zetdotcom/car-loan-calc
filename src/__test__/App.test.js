import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from 'App';

afterEach(cleanup);

it('renders', () => {
	const { asFragment } = render(<App />);
	expect(asFragment()).toMatchSnapshot();
});
