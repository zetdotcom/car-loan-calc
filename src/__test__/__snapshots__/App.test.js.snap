// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-fzqNJr eLbWnw"
  >
    <header
      class="sc-fzqBZW hkmDBn"
    >
      <h1
        class="sc-AxirZ eqtHGB"
      >
        Interest free car loan
      </h1>
    </header>
    <div
      class="sc-AxmLO evPyu"
    >
      <div
        class="sc-fzozJi dVvcpM"
      >
        <form>
          <div
            class="MuiPaper-root sc-AxhUy icrqyL MuiPaper-elevation3 MuiPaper-rounded"
          >
            <h2
              class="sc-AxiKw hsrZvz"
            >
              Start loan calculation
            </h2>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                data-shrink="true"
                for="price-input"
                id="price-input-label"
              >
                Vehicle price
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="price-input"
                  required=""
                  type="number"
                  value="10000"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-87 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-89 PrivateNotchedOutline-legendNotched-90"
                  >
                    <span>
                      Vehicle price *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                data-shrink="true"
                for="deposit-input"
                id="deposit-input-label"
              >
                Deposit
                <span
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-describedby="deposit-input-helper-text"
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="deposit-input"
                  required=""
                  type="number"
                  value="2000"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-87 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-89 PrivateNotchedOutline-legendNotched-90"
                  >
                    <span>
                      Deposit *
                    </span>
                  </legend>
                </fieldset>
              </div>
              <p
                class="MuiFormHelperText-root MuiFormHelperText-contained MuiFormHelperText-filled Mui-required"
                id="deposit-input-helper-text"
              >
                Deposit value must beat least 15% of price: £1,500.00
              </p>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
              format="dd/MM/yyyy"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                data-shrink="true"
                for="date-picker"
                id="date-picker-label"
              >
                Delivery date
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="date-picker"
                  readonly=""
                  type="text"
                  value="21/02/2020"
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-87 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-89 PrivateNotchedOutline-legendNotched-90"
                  >
                    <span>
                      Delivery date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <fieldset
              class="MuiFormControl-root MuiFormControl-marginNormal"
            >
              <legend
                class="MuiFormLabel-root"
              >
                Loan term
              </legend>
              <div
                aria-label="Loan term"
                class="MuiFormGroup-root sc-AxgMl jaInvp"
                role="radiogroup"
              >
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-131 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-134"
                        name="loan term"
                        type="radio"
                        value="1"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-147"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-148"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    1 year
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-131 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-132 Mui-checked MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-134"
                        name="loan term"
                        type="radio"
                        value="2"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-147 PrivateRadioButtonIcon-checked-149"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-148"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    2 years
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-131 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-134"
                        name="loan term"
                        type="radio"
                        value="3"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-147"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-148"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    3 years
                  </span>
                </label>
              </div>
            </fieldset>
            <br />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AxjAm dMOQXt MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Show results
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-fzozJi dotqUC"
      >
        <div>
          <h2
            class="sc-AxiKw hsrZvz"
          >
            Loan summary
          </h2>
          <h3
            class="sc-AxhCb hTTpNb"
            data-testid="totalLoanDisplay"
          >
            Total loan amount (incl. fees): £0.00
          </h3>
          <h3
            class="sc-AxhCb hTTpNb"
            data-testid="monthlyPaymentDisplay"
          >
            Monthly payment: £0.00
          </h3>
          <ul
            class="sc-AxheI eVldDd"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-fzoLsD komdMM"
    >
      <h2
        class="sc-AxiKw hsrZvz"
      >
        Cars affordable based on monthly payment
      </h2>
      <p>
        No results
      </p>
      <ul
        class="sc-fzpans tLglb"
      />
    </div>
  </div>
</DocumentFragment>
`;
